### ISSUE TITLE
Non-standard server ports stripped by Installer and in generated config.inc.php files

### ISSUE THREAD

**Issue by mattkirwan** [modxcms/revolution#12425]
Hi,

I've been having some issues with installing and then generally browsing a modx site while running through a vagrant machine.

The general practice when using a vagrant machine is to forward the machine ports 80 or 443 to 8080/8443 (this is by no means fixed though).

It seems that the core/config/config.inc.php file that is generated when installing is removing the port number from the url on every request (eg: 'Click Next' in the setup) when NOT using port 80 OR using a https connection. (around line 60 [I don't know how to link to files in github]).

Why is it doing this? It doesn't seem right to be forcing a port number on a modx instance - apart from causing problems with development machines, what if I wanted this on an internal network at a specific port?

And of course everytime setup is ran it's reset.
I can submit a pull request, but thought it may need a discussion as I'm probably missing a key component as to 'why'.

Cheers,

Matt

**Comment by wshawn** [comment]
Ports are typically established in the web server. For Apache:
http://docs.vagrantup.com/v2/getting-started/provisioning.html
and
http://docs.vagrantup.com/v2/networking/basic_usage.html
and 
http://docs.vagrantup.com/v2/networking/forwarded_ports.html
should help.

According to the last page the local computer is on 8080 while traffic via the nic is on port 80.

**Comment by OptimusCrime** [comment]
@wshawn That is really not answering the problem adressed here.

**Comment by wshawn** [comment]
I believe it does. It appears the OP is trying to access the server via the internal port instead of the default port.

His statement:
The general practice when using a vagrant machine is to forward the machine ports 80 or 443 to 8080/8443 (this is by no means fixed though).

This is reverse as I under stand it. Port 80 and 443 are redirected to the internal 8080 IP Addresses, not the other way around.

**Comment by OptimusCrime** [comment]
Hmm, not sure if I understand completly, but.

As Vagrant is a virtual machine, it forwards the MV's ports 80 to 8080, there the latter is accessed outsite the MV.

**Comment by mattkirwan** [comment]
Thank you for your responses. It is not a vagrant. Here is the start of the offending code:
https://github.com/modxcms/revolution/blob/master/setup/index.php#L73
(Note this is the setup/index but it's the same or very similar for the outputted config).

Which leads me to ask:
It seems that the core/config/config.inc.php file that is generated when installing is removing the port number from the url on every request (eg: 'Click Next' in the setup) when NOT using port 80 OR using a https connection ?

TLDR; The port is not been used in the setup scripts OR the produced config file (from said setup) if not running on port 80 or 443.

I hope that makes more sense, and again thank you for your responses.

Matt
